proseco/tests/test_jupiter.py

```diff
@@ -0,0 +1,441 @@
+            target_name="Jupiter 3",
+            raise_exc=True,
+        )
+    )
+    assert len(aca.acqs) == 8
+    assert len(aca.guides) == 5
+    assert len(aca.fids) == 3
+    names = ["id", "yang", "zang", "row", "col", "mag", "spoiler_score", "idx"]
+    assert all(name in aca.fids.colnames for name in names)
+    assert aca.target_name == "Jupiter 3"
+
+
+def test_get_jupiter_position_returns_table():
+    # Use a known date, duration, and a simple attitude (RA, Dec, Roll)
+    date = "2025:093:12:26:04.000"
+    duration = 1000  # seconds
+    att = Quat(q=[-0.43419701, -0.51408310, 0.33920339, 0.65736792])
+    out = jupiter.get_jupiter_position(date, duration, att)
+
+    # Should return a Table or None
+    assert isinstance(out, Table)
+    assert all(col in out.colnames for col in ["time", "row", "col"])
+    assert len(out) > 0
+
+    # Confirm the times in the table are reasonable for the date and duration
+    start_sec = CxoTime(date).secs
+    assert np.all((out["time"] >= start_sec) & (out["time"] <= start_sec + duration))
+
+
+def test_jupiter_distribution_check_1():
+    # Simulate jupiter_data crosses 0
+    jupiter_data = JupiterPositionTable({"row": [-10, 20]})
+    cand_guide_set = Table({"row": [-400, -300, 400, 300]})
+    # Should pass: at least two on each side
+    assert jupiter.jupiter_distribution_check(cand_guide_set, jupiter_data)
+    # Should fail: only with all on one side
+    cand_guide_set = Table({"row": [-400, -300, -200]})
+    assert not jupiter.jupiter_distribution_check(cand_guide_set, jupiter_data)
+
+
+def test_jupiter_distribution_check_2():
+    # Simulate jupiter_data all positive
+    jupiter_data = JupiterPositionTable({"row": [10, 20]})
+    cand_guide_set = Table({"row": [-400, -300, 400, 300]})
+    # Should pass: at least two on each side
+    assert jupiter.jupiter_distribution_check(cand_guide_set, jupiter_data)
+    # Should fail: only with all on one side
+    cand_guide_set = Table({"row": [400, 300, 200]})
+    assert not jupiter.jupiter_distribution_check(cand_guide_set, jupiter_data)
+
+
+def test_check_spoiled_by_jupiter():
+    # Simulate candidate stars and jupiter data
+    cand_stars = Table({"id": [1, 2, 3], "col": [10, 50, 100], "row": [0, 0, 0]})
+    jupiter_data = JupiterPositionTable({"col": [40, 60], "row": [0, 0]})
+    mask, rej = jupiter.check_spoiled_by_jupiter(cand_stars, jupiter_data)
+    # Only star with col=50 should be spoiled (within 15 pixels of Jupiter)
+    assert np.array_equal(mask, [False, True, False])
+    assert len(rej) == 1
+    assert rej[0]["id"] == 2
+
+
+def test_add_jupiter_as_spoilers():
+    date = "2025:220:12:00:00"
+    stars = StarsTable.empty()
+    jupiter_data = Table({"time": [CxoTime(date).secs], "row": [100], "col": [100]})
+    out = jupiter.add_jupiter_as_lots_of_acq_spoilers(date, stars, jupiter_data)
+    # Should add a new star with id=20 at Jupiter's position
+    assert len(out) > 1000
+    assert 1000 in out["id"]
```

Comment:
Just to be sure I made a plot.

<img width="326" height="325" alt="Image" src="https://github.com/user-attachments/assets/502bcc6b-1a19-4551-bc2e-027b10ea0099" />